optuna/visualization/_contour.py

```diff
@@ -87,15 +87,26 @@ def _get_contour_plot(study: Study, params: Optional[List[str]] = None) -> "go.F
                 raise ValueError("Parameter {} does not exist in your study.".format(input_p_name))
         sorted_params = sorted(list(set(params)))
 
+    padding_ratio = 0.05
     param_values_range = {}
     for p_name in sorted_params:
         values = [t.params[p_name] for t in trials if p_name in t.params]
-        param_values_range[p_name] = (min(values), max(values))
+        if _is_log_scale(trials, p_name):
+            padding = (math.log10(max(values)) - math.log10(min(values))) * padding_ratio
+            min_value = math.pow(10, math.log10(min(values)) - padding)
+            max_value = math.pow(10, math.log10(max(values)) + padding)
+        else:
+            padding = (max(values) - min(values)) * padding_ratio
+            min_value = min(values) - padding
+            max_value = max(values) + padding
+        param_values_range[p_name] = (min_value, max_value)
```

Comment:
Re-calculating max and min of `values` are not necessary. How about temporarily preserving them?
```suggestion
        max_value = max(values)
        min_value = min(values)
        if _is_log_scale(trials, p_name):
            padding = (math.log10(max_value) - math.log10(min_value)) * padding_ratio
            min_value = math.pow(10, math.log10(min_value) - padding)
            max_value = math.pow(10, math.log10(max_value) + padding)
        else:
            padding = (max_value - min_value) * padding_ratio
            min_value = min_value - padding
            max_value = max_value + padding
        param_values_range[p_name] = (min_value, max_value)
```